src/gui/MixerView.cpp

```diff
@@ -206,8 +206,11 @@ void MixerView::refreshDisplay()
 	// delete all views and re-add them
 	for (int i = 1; i<m_mixerChannelViews.size(); ++i)
 	{
-		chLayout->removeWidget(m_mixerChannelViews[i]);
-		m_racksLayout->removeWidget(m_mixerChannelViews[i]->m_effectRackView);
+		auto * mixerChannelView = m_mixerChannelViews[i];
+		chLayout->removeWidget(mixerChannelView);
+		m_racksLayout->removeWidget(mixerChannelView->m_effectRackView);
+
+		delete mixerChannelView;
```

Comment:
I think `deleteLater` should not always be used but rather in situations where it is needed (some scenarios are listed [here](https://stackoverflow.com/questions/22376298/when-to-use-deletelater)). In fact, I think it might even lead to complications when it is used in a context where a direct deletion works as well.

Calling `delete` will immediately unlink the `MixerChannelView` from it's parent and delete the instance. The link between the parent and the child likely was what lead to the problems where the `Fader` was repainted without a model. So if we'd call `deleteLater` instead of `delete` an event for later deletion will be put into the event queue. If there's an event in the queue that will be handled before the deletion event and which leads to a repaint then we'd still run into problem. This is not the case here but it shows that `deleteLater` might also lead to problems.

Reply:
> The link between the parent and the child likely was what lead to the problems where the Fader was repainted without a model.

Ideally, the code should be using some RAII mechanism at the highest level for the project tree. Reloading the project would imply deletion of some old `Project` object with a new one, so child objects are not left alive and bugs like this can never happen, but for now this works.